cmd/dockerd/config.go

```diff
@@ -59,6 +59,7 @@ func installCommonConfigFlags(conf *config.Config, flags *pflag.FlagSet) {
 	flags.IntVar(&maxConcurrentDownloads, "max-concurrent-downloads", config.DefaultMaxConcurrentDownloads, "Set the max concurrent downloads for each pull")
 	flags.IntVar(&maxConcurrentUploads, "max-concurrent-uploads", config.DefaultMaxConcurrentUploads, "Set the max concurrent uploads for each push")
 	flags.IntVar(&conf.ShutdownTimeout, "shutdown-timeout", defaultShutdownTimeout, "Set the default shutdown timeout")
+	flags.IntVar(&conf.NetworkDiagnose, "network-diagnose", -1, "TCP port number of the network diagnose server")
```

Comment:
Can we make the default `0` or is that a valid port?

Also, per our discussion; I think we wanted to make this a config-file only feature (i.e., only support setting this on `daemon.json`, but not through a flag. Not sure if a flag is required to make the config work (I _think_ we can do without), but if it's required, we can hide the flag perhaps?

Reply:
Also `s/diagnose server/diagnostics server/` ?

Reply:
@thaJeztah `0` means pick an available port automatically for me

let me add some bikeshed too :D 

could we maybe use `port` in the flag name ? such as `"metrics-addr"` instead of just `"metrics"` ? When I look at the current flag it looks like a bool to me.

Reply:
like network-diagnostic-port ?

Reply:
My reason for `0` was that it's the default value for an integer; while we do `-1` for other options, most we use it if `0` itself is a valid value.

Agree on adding `-port` (was thinking about that, but thought it may be too verbose)